File: TEAM B2 (Season 2)/Week 2/Musk/[스택_큐_기능개발].cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;


vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int isComplete = 0;
    
    while(isComplete < progresses.size()) {
        int temp = 0;
        
        for(int i=isComplete;i<progresses.size();i++) {
            if(progresses[i] >= 100 && isComplete == i) { 
                temp++;
                isComplete++;
            }
            progresses[i] += speeds[i];
        }
        
        if(temp > 0) { answer.push_back(temp); }
    }
    
    
    
    return answer;
}

// vector<int> solution(vector<int> progresses, vector<int> speeds) {
//     vector<int> answer;
//     int isComplete = 0;
    
//     while(isComplete < progresses.size()) {
//         int temp = 0;
        
//         for(int i=0;i<progresses.size();i++) {
//             if((progresses[i] < 100) && (progresses[i] + speeds[i] >= 100)) {
//                 temp++;
//                 isComplete++;
//             }
//             progresses[i] += speeds[i];
//         }
//         if(temp > 0) { answer.push_back(temp); }
//     }
//     return answer;
// }

File: TEAM B2 (Season 2)/Week 2/Ollie/[스택큐_기능개발].py

# 기능 개발
# https://school.programmers.co.kr/learn/courses/30/lessons/42586
from typing import List

def solution(progresses: List[int], speeds: List[int]) -> List[int]:
    if not progresses: return []
    workingDayStack = workingDayList(progresses, speeds)[::-1]
    result: List[int] = []
    tempWorkingDay = workingDayStack.pop()
    tempCount = 1
    while workingDayStack:
        currentWorkingDay = workingDayStack.pop()
        if tempWorkingDay < currentWorkingDay:
            result.append(tempCount)
            tempWorkingDay = currentWorkingDay
            tempCount = 1
        else:
            tempCount += 1
    result.append(tempCount)
    return result

import math
def workingDayList(progresses: List[int], speeds: List[int]) -> List[int]:
    return [ math.ceil((100-progress)/speed) for (progress, speed) in zip(progresses, speeds)]
    

def solution(progresses, speeds):
    import math
    days = [math.ceil((100-p)/s) for p, s in zip(progresses, speeds)]
    
    answer = []
    cnt, curr_max = 1, days[0]
    for day in days[1:]:
        if curr_max < day:
            answer.append(cnt)
            cnt = 1
            curr_max = day
        else:
            cnt += 1
    answer.append(cnt)
    return answer